lib/charms/traefik_k8s/v0/ingress_per_unit.py

```diff
@@ -246,30 +581,71 @@ def app_name(self):
 
         return None
 
-    @property
+    @cached_property
     def units(self):
-        """The remote units."""
-        return sorted(self._relation.units, key=lambda unit: unit.name)
+        """The remote units that have posted requests, i.e., those that have provided the name."""
+        ready_units = filter(self.is_unit_ready, self._relation.units)
+        return sorted(ready_units, key=lambda unit: unit.name)
+
+    def is_unit_ready(self, unit: Unit) -> bool:
+        """Returns whether the given unit has provided its name, model, host and port data."""
+        self._check_unit_belongs_to_relation(unit)
```

Comment:
Would it make more sense to have this return a bool instead of raising?
If not, then should probably add a `Raises:` statement.
But shouldn't `is_ready`  return a bool without raising regardless?

Reply:
@mmanciop what was the idea of making it raise? 
I see most of the methods which use _check_unit_belongs_to_relation return Optional[Trype], so wouldn't it make more sense to do:
```
if not self._unit_belongs_to_relation(unit):
    return None
```
or do we want the user to know that there's something deeply wrong with what they're trying to do?